core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -177,17 +178,18 @@ private FileSystemContext(Subject subject) {
    * Initializes the context. Only called in the factory methods and reset.
    *
    * @param masterInquireClient the client to use for determining the master
+   * @param enableUserMetricsCollection if enables the user metrics collection
    */
-  private synchronized void init(MasterInquireClient masterInquireClient) {
+  private synchronized void init(MasterInquireClient masterInquireClient,
+      boolean enableUserMetricsCollection) {
```

Comment:
instead of passing `enableUserMetricsCollection` as a boolean, could callers just set the Configuration property?

Reply:
hmm, I thought about that. The difference is that having a parameter here notifies the caller that this configuration should be considered.